File: android/src/main/kotlin/dev/hebei/mars_logging/XLogImpl.kt
package dev.hebei.mars_logging

class XLogImpl : XLogApi {
    override fun open(
        mode: AppenderModeApi,
        logsDir: String,
        cacheDir: String,
        cacheDays: Long,
        nameprefix: String,
        useConsole: Boolean,
        level: XLogLevelApi
    ) {
        XLog.open(mode.obj, logsDir, cacheDir, cacheDays.toInt(), nameprefix, useConsole, level.obj)
    }

    override fun flush(isSync: Boolean) {
        XLog.flush(isSync)
    }

    override fun close() {
        XLog.close()
    }

    override fun verbose(tag: String, message: String) {
        XLog.verbose(tag, message)
    }

    override fun debug(tag: String, message: String) {
        XLog.debug(tag, message)
    }

    override fun info(tag: String, message: String) {
        XLog.info(tag, message)
    }

    override fun warning(tag: String, message: String) {
        XLog.warning(tag, message)
    }

    override fun error(tag: String, message: String) {
        XLog.error(tag, message)
    }

    override fun fatal(tag: String, message: String) {
        XLog.fatal(tag, message)
    }
}

val AppenderModeApi.obj: AppenderMode
    get() = when (this) {
        AppenderModeApi.ASYNC -> AppenderMode.ASYNC
        AppenderModeApi.SYNC -> AppenderMode.SYNC
    }

val XLogLevelApi.obj: XLogLevel
    get() = when (this) {
        XLogLevelApi.ALL -> XLogLevel.ALL
        XLogLevelApi.VERBOSE -> XLogLevel.VERBOSE
        XLogLevelApi.DEBUG -> XLogLevel.DEBUG
        XLogLevelApi.INFO -> XLogLevel.INFO
        XLogLevelApi.WARNING -> XLogLevel.WARNING
        XLogLevelApi.ERROR -> XLogLevel.ERROR
        XLogLevelApi.FATAL -> XLogLevel.FATAL
        XLogLevelApi.NONE -> XLogLevel.NONE
    }

File: darwin/Classes/XLogImpl.swift
//
//  XLogImpl.swift
//  mars_logging
//
//  Created by 闫守旺 on 2025/3/19.
//

import Foundation

class XLogImpl: XLogApi {
    func open(mode: AppenderModeApi, logsDir: String, cacheDir: String, cacheDays: Int64, nameprefix: String, useConsole: Bool, level: XLogLevelApi) throws {
        XLog.open(mode.obj, logsDir: logsDir, cacheDir: cacheDir, cacheDays: Int32(cacheDays), nameprefix: nameprefix, useConsole: useConsole, level: level.obj)
    }
    
    func flush(isSync: Bool) throws {
        XLog.flush(isSync)
    }
    
    func close() throws {
        XLog.close()
    }
    
    func verbose(tag: String, message: String) throws {
        XLog.verbose(tag, message: message)
    }
    
    func debug(tag: String, message: String) throws {
        XLog.debug(tag, message: message)
    }
    
    func info(tag: String, message: String) throws {
        XLog.info(tag, message: message)
    }
    
    func warning(tag: String, message: String) throws {
        XLog.warning(tag, message: message)
    }
    
    func error(tag: String, message: String) throws {
        XLog.error(tag, message: message)
    }
    
    func fatal(tag: String, message: String) throws {
        XLog.fatal(tag, message: message)
    }
}

extension AppenderModeApi {
    var obj: AppenderMode {
        switch self {
        case .async:
            return .async
        case .sync:
            return .sync
        }
    }
}

extension XLogLevelApi {
    var obj: XLogLevel {
        switch self {
        case .all:
            return .all
        case .verbose:
            return .verbose
        case .debug:
            return .debug
        case .info:
            return .info
        case .warning:
            return .warning
        case .error:
            return .error
        case .fatal:
            return .fatal
        case .none:
            return .none
        }
    }
}
